src/rgw/rgw_rados.cc

```diff
@@ -5071,12 +5080,30 @@ int RGWRados::create_pool(rgw_bucket& bucket)
   if (ret < 0)
     return ret;
 
+  if (! bucket.name_space.empty()) {
+    librados::IoCtx io_ctx;
+    ret = rad->ioctx_create(pool.c_str(), io_ctx);
+    if (ret) {
+      return ret;
+    }
+    io_ctx.set_namespace(bucket.name_space);
```

Comment:
yes, i missed this one, realized after creating the PR, will add this functionality. Subsequent calls to add objects to in a different namespace in the same pool will result in the 'default' namespace, which we dont want.